src/parsers/json/Parser.cpp

```diff
@@ -482,14 +500,17 @@ const ElemR* Parser::boxToElemGroup(Mesh::Unstructured& mesh, const std::string&
 {
     BoxR3 box = strToBox(line);
     std::unique_ptr<ElemR> elem;
-    if (box.isVolume()) {
-        elem = std::make_unique<HexR8>(mesh.coords(), ElemId(0), box);
+
+    auto posVec = box.getPos();
+
+    if (box.isVolume()) {    
+        elem = std::make_unique<HexR8>(ElemId(), addAngGetCoordView(mesh.coords(), posVec, Geometry::HexR8::sizeOfCoordinates).data());
```

Comment:
can we return directly?

Reply:
Up to you, that is a pure-styling thing.